ordered_model/management/commands/reorder_model.py

```diff
@@ -0,0 +1,70 @@
+from django.apps import apps
+from django.core.management import BaseCommand, CommandError
+
+from ordered_model.models import OrderedModelBase
+
+
+def get_order_with_respect_to(model):
+    if not model.order_with_respect_to:
+        return None
+    if isinstance(model.order_with_respect_to, str):
+        return model.order_with_respect_to
+    assert (
+        len(model.order_with_respect_to) <= 1
+    ), "re-ordering with more than 1 fields in order_with_respect_to is not supported"
+    return model.order_with_respect_to[0]
+
+
+class Command(BaseCommand):
+    help = "Re-do the ordering of a certain Model"
+
+    def add_arguments(self, parser):
+        parser.add_argument("model_name", type=str, nargs="+")
+
+    def handle(self, *args, **options):
+        """
+        Sometimes django-ordered-models ordering goes wrong, for various reasons,
+        try re-ordering to a working state.
+        """
+        self.verbosity = options["verbosity"]
+        for model_name in options["model_name"]:
+            model = apps.get_model(model_name)
+            if not issubclass(model, OrderedModelBase):
+                raise CommandError(
+                    "{} does not inherit from OrderedModel or OrderedModelBase".format(
+                        str(model)
+                    )
+                )
+
+            self.reorder(model)
+
+    def reorder(self, model):
+        if model.order_with_respect_to:
+            order_with_respect_to = get_order_with_respect_to(model)
+            rel_kwargs = {"{}__isnull".format(order_with_respect_to): False}
+            relation_to_list = (
+                model.objects.order_by(order_with_respect_to)
+                .values_list(order_with_respect_to, flat=True)
+                .filter(**rel_kwargs)
+                .distinct()
+            )
+            for relation_to in relation_to_list:
+                kwargs = {order_with_respect_to: relation_to}
+                self.reorder_queryset(model.objects.filter(**kwargs))
+        else:
+            self.reorder_queryset(model.objects.all())
+
+    def reorder_queryset(self, queryset):
+        model = queryset.model
+        order_field_name = model.order_field_name
+
+        for order, obj in enumerate(queryset):
```

Comment:
We should do bulk_update here. And also to wrap it in `atomic`

Reply:
Why `bulk_update` as the expected number of changes is very small.
I've wrapped the changes within a `with transaction.atomic():` block

Reply:
We don't know the expected number. It could be thousands

Reply:
Sure, in theory it could be. But the likelihood is that if you are re-ordering objects by hand with our Admin tool, then it's going to be pretty manageably small dataset. I'd rather revisit this on the occasion we get a bug report suggests it too slow.

Reply:
The patch to use bulk_update is small and straightforward, however the api was introduced in Django 2.2 and we still support and test against Django 2.0 and Django 2.1 so I suggest we hold off until we deprecate those versions.

```$ git diff
diff --git a/ordered_model/management/commands/reorder_model.py b/ordered_model/management/commands/reorder_model.py
index eb22830..9670cd6 100644
--- a/ordered_model/management/commands/reorder_model.py
+++ b/ordered_model/management/commands/reorder_model.py
@@ -73,7 +73,7 @@ class Command(BaseCommand):
     def reorder_queryset(self, queryset):
         model = queryset.model
         order_field_name = model.order_field_name
-
+        objs = []
         for order, obj in enumerate(queryset):
             if getattr(obj, order_field_name) != order:
                 if self.verbosity:
@@ -83,4 +83,6 @@ class Command(BaseCommand):
                         )
                     )
                 setattr(obj, order_field_name, order)
-                obj.save()
+                objs.append(obj)
+            queryset.bulk_update(objs, [order_field_name])
+
```